Issue #12

gorshkovayulia opened the pull request:
Don't go into infinite loop in case of bidirectional relationships
**Note:** not sure about naming of some new tests --> will be improved a bit later.

gorshkovayulia committed:
Issue #12
Don't go into infinite loop in case of bidirectional relationships

gorshkovayulia committed:
Issue #13
Exclude fields of child objects

gorshkovayulia committed:
Issue #12
- fix case when internal objects of actual core object and expected core object are the same or both null

ctapobep commented on core/src/main/java/io/elsci/assertreflectionequals/ReflectionAssert.java:
```diff
@@ -61,39 +63,47 @@ private void assertReflectionEquals(Deque<Class<?>> fullPath, Object expectedObj
                                        Object expectedObject, Object actualObject) {
-        assertReflectionEquals(fullPath, ReflectionUtil.get(expectedField, expectedObject),
-                ReflectionUtil.get(expectedField, actualObject));
+        if (objects.contains(ReflectionUtil.get(expectedField, expectedObject)) ||
```
Pushed to `stas-failing-test` the test case that demonstrate that this logic doesn't work properly.

gorshkovayulia committed:
1. Fix checking objects that contain internal objects;
2. Fix building error messages in case internal objects.

gorshkovayulia commented:
The working version with bug fixes were pushed -> it's not ready for review.
Tests and refactored code will be pushed soon.

gorshkovayulia committed:
Refactor code + add tests + fix minor bugs

gorshkovayulia commented:
Now it's ready for review.
Note: some tests are failed because of problem with checking of full error message containing hash code of object.

gorshkovayulia committed:
Remove unused class

gorshkovayulia committed:
1. Add IdentityHashSet implementation;
2. Start using of IdentityHashSet + rework assertReferencesEqual method;
3. Rework handling of case when two objects are not objects of the same class: add separate method for building error;
4. Remove extra dot sign from error message when no field names are used;
5. Fix tests where error message contains hash code of object

gorshkovayulia committed:
#12 Refactor assertReferencesEqual method

gorshkovayulia committed:
#12: Cosmetic improvement + uncomment some tests

gorshkovayulia committed:
#12
1. Rename test

gorshkovayulia committed:
#13 Refactor getProperFields() method
